### PR TITLE
Site-wide Announcement System

### PR DESCRIPTION
**Pull request by NuckChorris**

### CODE REVIEW

File: app/models/site_announcement.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class SiteAnnouncement < ApplicationRecord
+  include WithActivity
+
+  belongs_to :user, required: true
+
+  validates :title, presence: true
+
+  def stream_activity
+    Feed::SiteAnnouncementsFeed.global.activities.new
```
Comment (trmcnvn) [inline app/models/site_announcement.rb]: Shouldn't this be a per-user notification feed, so we can mark announcements as seen?

Comment (trmcnvn) [reply app/models/site_announcement.rb]: Apparently, it is. nvm :nerd_face:

**Comment by trmcnvn** [comment]
```
User.first.site_announcements_feed.setup!
# => {"target"=>["You can only follow flat feeds, not feeds of type notification"]}:
```

**Comment by NuckChorris** [comment]
It should be a flat feed, we don't need agg— fuck notification feeds are aggregated only

Okay we'll need a second feed then

**Pull request merged**